nucypher/crypto/utils.py

```diff
@@ -40,10 +45,185 @@ def construct_policy_id(label: bytes, stamp: bytes) -> bytes:
     return keccak_digest(label + stamp)
 
 
-def canonical_address_from_umbral_key(public_key: Union[UmbralPublicKey, SignatureStamp]) -> bytes:
-    if isinstance(public_key, SignatureStamp):
-        public_key = public_key.as_umbral_pubkey()
-    pubkey_compressed_bytes = public_key.to_bytes(is_compressed=True)
-    eth_pubkey = EthKeyAPI.PublicKey.from_compressed_bytes(pubkey_compressed_bytes)
+def canonical_address_from_umbral_key(public_key: UmbralPublicKey) -> bytes:
+    pubkey_raw_bytes = get_coordinates_as_bytes(public_key)
+    eth_pubkey = EthKeyAPI.PublicKey(pubkey_raw_bytes)
     canonical_address = eth_pubkey.to_canonical_address()
     return canonical_address
+
+
+def get_coordinates_as_bytes(point: Union[Point, UmbralPublicKey, SignatureStamp],
```

Comment:
I think it was removed in #2699

Reply:
`canonical_address_from_umbral_key()` is also modified to not require it. In fact, in #2612 `crypto/utils.py` got pretty small. See https://github.com/fjarri/nucypher/blob/7f36abde216d5b1e751e10a95d6f382d7e296474/nucypher/crypto/utils.py